src/paho/mqtt/client.py

```diff
@@ -897,7 +899,13 @@ def publish(self, topic, payload=None, qos=0, retain=False):
             message.dup = False
 
             self._out_message_mutex.acquire()
+
+            if self._max_out_messages > 0 and len(self._out_messages) >= self._max_out_messages:
+                self._out_message_mutex.release()
+                return (MQTT_ERR_NOMEM, local_mid)
```

Comment:
What's the appropriate error code. Could you please give me a name? I'm not an English speaker.